### PR TITLE
Did model and base to-do UI

### PR DESCRIPTION
**Pull request by yashasvi-karri**

### CODE REVIEW

File: Vineyard/Vineyard/ViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//
+//  Created by Yashasvi Karri on 9/12/24.
+//
+
+import SwiftUI
+
+@Observable
+class ViewModel {
+    var user: Person = Person(name: "Andrew", age: 21, participatingGroups: [
+        Group(people: [Person(name: "Yash", age: 21)], groupName: "New Group Name", yearlyResolution: "eat healthy", resolutions: [Resolution(timeBound: TimeBound.day, name: "String", goal: 9.0, freq: 10)]),
```
Comment (sankaet8) [inline Vineyard/Vineyard/ViewModel.swift:13]: Can you think of any way in which we can ensure that the Person and the Group both have the exact same people in it. For example here we can get into an issue with Andrew participating in the groups but the Groups not holding Andrew in the people.

**Review by sankaet8 — APPROVED** [review]
Really good job overall! I appreciate the fast turnaround time with this issue. I will make some more tickets to make some enhancements to this good work!

**Comment by sankaet8** [comment]
Good catch @vishneshjr :)

**Comment by sankaet8** [comment]
Also @yashasvi-karri, please add a screen recording on your next PRs

File: Vineyard/Vineyard/ResolutionView.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+//  Vineyard
+//
+//  Created by Yashasvi Karri on 9/12/24.
+//
+
+import SwiftUI
+
+struct ResolutionView: View {
+    @Environment(ViewModel.self) private var viewModel
+    var group: Group
```
Comment (rahulon12) [inline Vineyard/Vineyard/ResolutionView.swift]: nit: 
```suggestion
    let group: Group
```

File: Vineyard/Vineyard/ViewModel.swift (hunk 1 above)
Comment (rahulon12) [reply Vineyard/Vineyard/ViewModel.swift:13]: I know that `CoreData` and/or `SwiftData` make this kind of a relational database easy to implement. Wasn't able to find any clear resources on how Firebase does this. I found this: https://medium.com/firebase-tips-tricks/how-to-secure-many-to-many-relationships-in-firestore-d19f972fd4d3.

cc @gtiosclub/vineyard-officers

File: Vineyard/Vineyard/ResolutionView.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                        }
+                    })
+                }
+            }
+        }
+    }
+}
+
+#Preview {
+    ResolutionView(group: Group(people:[Person(name: "BRuh", age: 101)], groupName: "", yearlyResolution: ""))
```
Comment (rahulon12) [inline Vineyard/Vineyard/ResolutionView.swift]:
```suggestion
    ResolutionView(group: Group(people:[Person(name: "BRuh", age: 101)], groupName: "", yearlyResolution: ""))
                .environment(ViewModel())
```

Same for the one in `ContentView.swift`

**Pull request merged**